utils/databake/src/converter.rs

```diff
@@ -0,0 +1,261 @@
+//! Helper structs to generate a different output than given input.
+//!
+//! This allows e.g. to store a `String` but output a `&'static str`.
+
+use crate::{Bake, BakeSize, CrateEnv};
+use core::marker::PhantomData;
+use core::ops::{Deref, DerefMut};
+use proc_macro2::TokenStream;
+use quote::quote;
+
+/// Let [`Bake`] output a `&'static str` for anything that can be `AsRef<str>`.
+///
+/// It can be helpful if the type needs to be constructed (e.g. [`String`], [`Cow<str>`](std::borrow::Cow)) but should result in a `&'static str`.
+///
+/// This requires that the crate using the generated data needs a different struct.
+///
+/// ```
+/// use databake::{Bake, converter::AsStaticStr};
+///
+/// #[derive(Bake)]
+/// #[databake(path = my_crate)]
+/// struct Data {
+///     number: usize,
+///     string: AsStaticStr<String>, // can be written as StringAsStaticStr
+/// }
+///
+/// let data = Data {
+///     number: 6,
+///     string: 6.to_string().into()
+/// };
+///
+/// assert_eq!(
+///     data.bake(&Default::default()).to_string(),
+///     r#"my_crate :: Data { number : 6usize , string : "6" , }"#
+/// );
+///
+/// mod my_crate {
+///     struct Data {
+///         number: usize,
+///         string: &'static str,
+///     }
+/// }
+/// ```
+#[repr(transparent)]
+pub struct AsStaticStr<T>(pub T);
```

Comment:
given that the whole purpose of this wrapper is the `Bake` implementation, I'd add the `T: AsRef<str>` bound here.

Reply:
I've also added it to AsRefSlice